Automattic/wp-calypso
# Adds unit testing to the Blaze Dashboard app

## Proposed Changes

This PR adds unit testing to the Blaze Dashboard app. I also refactor the files to use typescript.

## Testing Instructions

1. Check out this branch
2. Go to `/apps/blaze-dashboard`
3. Execute `yarn test:js` and check that all the unit tests run successfully

**To test the app:**

1. Check out this Jetpack branch: https://github.com/Automattic/jetpack/pull/30103
2. In the console, go to `apps/blaze-dashboard` and execute:
```
yarn install
BLAZE_DASHBOARD_PACKAGE_PATH=/path/to/jetpack/projects/packages/blaze yarn dev
```
**Note:** Change the jetpack path to the correct folder installation.

3. Build the Jetpack Blaze package: `pnpm jetpack build packages/blaze`
4. Open your Jurrasic Tube tunnel and then in the browser go to the wp-admin page for your local site
**Note:** You need to enable this filter jetpack_blaze_dashboard_enable inside your local Wordpress installation.
You can ping me if you don't know how to do that.

5. Click on Tools->Advertising
6. Verify that the Dashboard is displayed correctly.

## Pre-merge Checklist
